### PR TITLE
Replace eval with asteval

### LINKED ISSUE #96: Avoid using unvalidated strings in eval
I've noticed that the density equations are evaluated by using just `eval(self.density_equation)`. Strings passed into `eval` should really be validated - particularly since the `material.py` module imports `os`, so in principle any `os` command could be executed. For example, if I create a json file containing the following I see that creating BadMat with some temperature in K will run the `os.system` command that I have provided.

```
{
    "BadMat": {
        "chemical_equation": "H2O",
        "density_equation": "os.system('ls')",
        "density_unit": "g/cm3",
        "temperature_dependant": true,
        "percent_type": "ao"
    }
}
```

There are a few tools for handling this, such as `asteval`, or `numexpr`. Note that this may need the string to be parsed to replace the temperature (or other material attributes) in the density_equation.

### PR DESCRIPTION
**Pull request by DanShort12**
The useage of a bare `eval` call is not good security practice within Python, particularly as we are accepting external input in the form of JSON files, which then contain text that will be executed. This PR proposes to use the [`asteval`](http://newville.github.io/asteval/) package to only run sanitised inputs. This package supports various `math`/`numpy` functions and allows symbols to explicitly be added in e.g. for the `PropsSI` functionality, so should fulfill what is needed while restricting the allowable external input.

Closes #96

### CODE REVIEW

**Comment by shimwell** [comment]
Thanks so much for this @DanShort12 I use eval in a few other projects so I shall tidy up as you have demonstrated.
Much appreciated

**Comment by shimwell** [comment]
Looks like this might be best to re-target this to develop then from there to main

Screenshot from 2020-10-09 11-32-10

File: requirments.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
+asteval==0.9.19
```
Comment (shimwell) [inline requirments.txt:1]: Does it need to be pinned to a particular version or can we saw newer than 0.9
```suggestion
asteval>=0.9.19
```

Comment (shimwell) [reply requirments.txt:1]: this is optional comment, feel free to ignore

**Comment by shimwell** [comment]
I think this can be merged with or without that optional correction

**Comment by DanShort12** [comment]
> Looks like this might be best to re-target this to develop then from there to main
> 
> Screenshot from 2020-10-09 11-32-10

Ah, yes, definitely agree. TBH, hadn't spotted that this repo was using the standard git flow but happy to see that's in place already :)

File: requirments.txt (hunk 1 above)
Comment (DanShort12) [reply requirments.txt:1]: I generally prefer to be specific in requirements.txt (tells you how to recreate a complete, working environment) and flexible in setup.py (tells pip what dependencies need to be picked up if e.g. installing in some other environment). PyPA have a [few notes](https://packaging.python.org/discussions/install-requires-vs-requirements/) on the matter.

**Pull request merged**